### PR TITLE
Support Gemini LLM and Embeddings on VertexAI https://github.com/llm-tools/embedJs/issues/46

### PR DESCRIPTION
**Pull request by tsibg**
Added support for Gemini model and embeddings on VertexAI. 
This PR also provides example and documentation on how to use, including updates to the main README.

Implemented with:
- Langchain integration for LLMS on VertexAI: https://js.langchain.com/docs/integrations/llms/google_vertex_ai
- Langchain's community integration (Experimental) for `textembedding-gecko` on VertexAI: https://js.langchain.com/docs/modules/data_connection/experimental/multimodal_embeddings/google_vertex_ai

Related Issue:
https://github.com/llm-tools/embedJs/issues/46

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -452,6 +454,36 @@ const ragApplication = await new RAGApplicationBuilder()
 
 You can read more about the various models provided by Anthropic [here](https://docs.anthropic.com/claude/docs/models-overview).
 
+
+## Vertex AI
```
Comment (adhityan) [inline README.md:458]: I will make a few modifications to the Readme documentation post merging the PR

File: README.md [hunk 2]
```diff
@@ -546,6 +578,24 @@ await new RAGApplicationBuilder()
 .setEmbeddingModel(new CohereEmbeddings())
 ```
 
+
+
+## Vertex AI Embeddings
+
+Embedding model `textembedding-gecko` with 768 dimensions on [VertexAI](https://cloud.google.com/vertex-ai/generative-ai/docs/embeddings/get-text-embeddings).
```
Comment (adhityan) [inline README.md:585]: We actually name the embedding to the actual model, not the provider (this allows us to use different classes for each model from a single provider). Could you rename the embedding model to `GeckoEmbedding` ?

File: package.json [hunk 3]
```diff
@@ -56,11 +56,13 @@
         "@langchain/cohere": "^0.0.8",
         "@langchain/community": "^0.0.52",
         "@langchain/core": "^0.1.60",
+        "@langchain/google-vertexai": "^0.0.14",
         "@langchain/mistralai": "^0.0.19",
         "@langchain/openai": "^0.0.28",
         "axios": "^1.6.8",
         "confluence.js": "^1.7.4",
         "debug": "^4.3.4",
+        "google-auth-library": "^8.9.0",
```
Comment (adhityan) [inline package.json:65]: I would move this to optional dependency like what we have with the vector DBs. I will do this post merge, nothing for you to do here.

**Comment by adhityan** [comment]
Thanks for sending in the PR; it's quite well written. There is one actionabale change that I think should be done pre-merge -

1. Rename the embedding model to reflect the actual model

Apart from that, there is an update I would like to do on the dependency front. I will do this and publish a new version post merge. What do you think about the change above?

**Comment by tsibg** [comment]
> Thanks for sending in the PR; it's quite well written. There is one actionabale change that I think should be done pre-merge -
> 
> 1. Rename the embedding model to reflect the actual model
> 
> Apart from that, there is an update I would like to do on the dependency front. I will do this and publish a new version post merge. What do you think about the change above?

Thanks for the review.
1. Makes sense. Will need 1-2days to get back on this.
2. The mentioned dependency  comes from the documentation for the Embeddings' module on Langchain. Let me double check if it is really needed and how the module is using it and will update you on this matter.

File: README.md (hunk 2 above)
Comment (tsibg) [reply README.md:585]: Refactored with https://github.com/llm-tools/embedJs/pull/47/commits/605fbd9029c8ef05bec58edddceb956a2ba74070 
Also updated the documentation to follow the changes.

File: package.json (hunk 3 above)
Comment (tsibg) [reply package.json]: I removed the dependency from package.json as https://github.com/llm-tools/embedJs/pull/47/commits/e41762fbc38b15ccf4dbf623939bb2ae810a9d58 .

**Comment by tsibg** [comment]
@adhityan Just to summarize the changes:

1. Renamed the model to GeckoEmbedings and updated the corresponding example and READMEs
2. Bit tricky. I tried to follow the dependencies of the related LangChain modules and it differs for the Model and for the Embeddings as follows:
- The Model already adds it as peer dependency: `@langchain/google-vertexai` -> `@langchain/google-gauth` -> `google-auth-library`
- The Embedding *does not* install it - it relies on experimental module, part of `@langchain/community`. However it needs it.

tl;dr I suppose that we can safely skip it for now. We already have installed `@langchain/google-vertexai` (and its peer dependencies). 
--
Please, take a look at the changes, it should be ready for merge.

**Comment by adhityan** [comment]
Thank you. LGTM. I will merge it now. Changes will be published to the next library version by end of week.
